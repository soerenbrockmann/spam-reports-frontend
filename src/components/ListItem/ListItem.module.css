html,
body {
    font-size: 10px;
}
h1 {
    font-size: 2rem;
}
.button {
    padding: 1rem;
    font-size: 3rem;
    width: 50%;
}
.gridContainer {
    display: grid;
    grid-template-columns: 33.33% 33.33% 33.33%;
    background-color: #2196f3;
    padding: 0.3rem;
}
.gridItem {
    background-color: rgba(255, 255, 255, 0.8);
    border: 0px solid rgba(0, 0, 0, 0.8);
    padding: 0.2rem;
    font-size: 1rem;
}
.gridItemRight {
    text-align: end;
}
