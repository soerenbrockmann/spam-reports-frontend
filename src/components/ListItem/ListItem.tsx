import React from 'react';
import { Button } from '../Button';
import styles from './ListItem.module.css';

interface ListItemProps {
    id: string;
    state: string;
    reportType: string;
    message: string;
    onBlockClick: Function;
    onResolveClick: Function;
}

enum Buttons {
    BLOCK = 'Block',
    RESOLVE = 'Resolve',
}

enum Labels {
    ID = 'Id:',
    TYPE = 'Type:',
    STATE = 'State:',
    MESSAGE = 'Message:',
    DETAILS = 'Details',
}

enum Url {
    URL = 'http://localhost:3000',
}

export const ListItem = ({
    id,
    state,
    reportType,
    message,
    onBlockClick,
    onResolveClick,
}: ListItemProps): JSX.Element => {
    const { gridContainer, gridItem, gridItemRight } = styles;
    return (
        <div>
            <div className={gridContainer}>
                <div className={gridItem}>{`${Labels.ID} ${id}`}</div>
                <div className={gridItem}>{`${Labels.TYPE} ${reportType}`}</div>
                <div className={`${gridItem} ${gridItemRight}`}>
                    <Button
                        id={id}
                        name={Buttons.BLOCK}
                        clickHandler={onBlockClick}
                    />
                </div>
                <div className={gridItem}>{`${Labels.STATE} ${state}`}</div>
                <div className={gridItem}>{`${Labels.MESSAGE} ${message}`}</div>
                <div className={`${gridItem} ${gridItemRight}`}>
                    <Button
                        id={id}
                        name={Buttons.RESOLVE}
                        clickHandler={onResolveClick}
                    />
                </div>
                <div className={gridItem}>
                    <a href={Url.URL}>{Labels.DETAILS}</a>
                </div>
                <div className={gridItem}></div>
                <div className={gridItem}></div>
            </div>
        </div>
    );
};
