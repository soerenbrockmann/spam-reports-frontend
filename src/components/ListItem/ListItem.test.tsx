import React from 'react';
import renderer from 'react-test-renderer';
import { ListItem } from './ListItem';

describe('<ListItem>', () => {
    it('renders correctly', () => {
        const listItemProps = {
            id: 'dummyId',
            state: 'dummyState',
            reportType: 'dummyReportType',
            message: 'dummyMessage',
            onBlockClick: jest.fn,
            onResolveClick: jest.fn,
        };
        const component = renderer.create(<ListItem {...listItemProps} />);
        expect(component).toMatchSnapshot();
    });
});
