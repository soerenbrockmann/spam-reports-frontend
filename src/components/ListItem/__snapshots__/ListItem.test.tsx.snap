// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem> renders correctly 1`] = `
<div>
  <div
    className="gridContainer"
  >
    <div
      className="gridItem"
    >
      Id: dummyId
    </div>
    <div
      className="gridItem"
    >
      Type: dummyReportType
    </div>
    <div
      className="gridItem gridItemRight"
    >
      <button
        className="button"
        id="dummyId"
        onClick={[Function]}
      >
        Block
      </button>
    </div>
    <div
      className="gridItem"
    >
      State: dummyState
    </div>
    <div
      className="gridItem"
    >
      Message: dummyMessage
    </div>
    <div
      className="gridItem gridItemRight"
    >
      <button
        className="button"
        id="dummyId"
        onClick={[Function]}
      >
        Resolve
      </button>
    </div>
    <div
      className="gridItem"
    >
      <a
        href="#"
      >
        Details
      </a>
    </div>
    <div
      className="gridItem"
    />
    <div
      className="gridItem"
    />
  </div>
</div>
`;
